### PR TITLE
Make builtin auth and secret plugins buildable

### PR DESCRIPTION
**Pull request by tyrannosaurus-becks**
This PR makes all builtin auth and secret backends buildable. In the directory of each, it adds a `cmd/:plugin-name` directory, with the plugin name corresponding with how it appears in the Vault CLI.

For example, `vault/builtin/credential/aws` gains `vault/builtin/credential/aws/cmd/aws/main.go`.

To build it as a plugin, you'd simply go into the `vault/builtin/credential/aws/cmd/aws` and run `$ go build`.

### CODE REVIEW

File: builtin/logical/database/cmd/database/main.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package main
```
Comment (briankassouf) [inline builtin/logical/database/cmd/database/main.go]: Database won't be able to be run as a plugin so we shouldn't offer a main file for it

Comment (tyrannosaurus-becks) [reply builtin/logical/database/cmd/database/main.go]: Oh right! Will remove it.

File: builtin/credential/github/cmd/github/main.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package main
```
Comment (kalafut) [inline builtin/credential/github/cmd/github/main.go]: This one doesn't build. I'm getting:

```
../../path_login.go:217:43: client.Organizations.ListUserTeams undefined (type *github.OrganizationsService has no field or method ListUserTeams)
```

The rest build fine though.

Comment (tyrannosaurus-becks) [reply builtin/credential/github/cmd/github/main.go]: I just merged in master, did `go build`, and confirmed that it's fixed.

**Pull request merged**